public/app/core/internationalization/constants.ts

```diff
@@ -10,61 +11,96 @@ export const PSEUDO_LOCALE = 'pseudo-LOCALE';
 
 export const DEFAULT_LANGUAGE = ENGLISH_US;
 
-interface LanguageDefinitions {
+export type LocaleFileLoader = () => Promise<ResourceKey>;
+
+export interface LanguageDefinition<Namespace extends string = string> {
   /** IETF language tag for the language e.g. en-US */
   code: string;
 
   /** Language name to show in the UI. Should be formatted local to that language e.g. Français for French */
   name: string;
 
   /** Function to load translations */
-  loader: () => Promise<ResourceKey>;
+  loader: Record<Namespace, LocaleFileLoader>;
 }
 
-export const LANGUAGES: LanguageDefinitions[] = [
+export const LANGUAGES: LanguageDefinition[] = [
   {
     code: ENGLISH_US,
     name: 'English',
-    loader: () => import('../../../locales/en-US/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/en-US/grafana.json'),
+    },
   },
 
   {
     code: FRENCH_FRANCE,
     name: 'Français',
-    loader: () => import('../../../locales/fr-FR/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/fr-FR/grafana.json'),
+    },
   },
 
   {
     code: SPANISH_SPAIN,
     name: 'Español',
-    loader: () => import('../../../locales/es-ES/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/es-ES/grafana.json'),
+    },
   },
 
   {
     code: GERMAN_GERMANY,
     name: 'Deutsch',
-    loader: () => import('../../../locales/de-DE/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/de-DE/grafana.json'),
+    },
   },
 
   {
     code: CHINESE_SIMPLIFIED,
     name: '中文（简体）',
-    loader: () => import('../../../locales/zh-Hans/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/zh-Hans/grafana.json'),
+    },
   },
 
   {
     code: BRAZILIAN_PORTUGUESE,
     name: 'Português Brasileiro',
-    loader: () => import('../../../locales/pt-BR/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/pt-BR/grafana.json'),
+    },
   },
-];
+] satisfies Array<LanguageDefinition<'grafana'>>;
 
 if (process.env.NODE_ENV === 'development') {
   LANGUAGES.push({
     code: PSEUDO_LOCALE,
     name: 'Pseudo-locale',
-    loader: () => import('../../../locales/pseudo-LOCALE/grafana.json'),
+    loader: {
+      grafana: () => import('../../../locales/pseudo-LOCALE/grafana.json'),
+    },
   });
 }
 
+// Optionally load enterprise locale extensions, if they are present.
+// It is important that this happens before NAMESPACES is defined so it has the correct value
+const extensionRequireContext = require.context('../../', true, /app\/extensions\/locales\/localeExtensions/);
```

Comment:
What are the pros of using require.context here?

Reply:
It allows for an optional import that won't fail if the file isn't present (like when building OSS).